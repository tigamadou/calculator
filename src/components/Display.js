import React from 'react'
import PropTypes from 'prop-types';

const Display = (props) => {
    return (
        <div className="display">
            {props.result}
        </div>
    )
}

Display.defaultProps = {
    result: 0
}
Display.propTypes = {
    result: PropTypes.string
}

export default Display