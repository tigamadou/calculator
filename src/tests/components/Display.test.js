import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Display from '../../components/Display';

describe('Display compoent tests', () => {
  test('It renders Display component with the text 0', () => {
    render(<Display />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('0')).toBeInTheDocument;
  });

  test('It match with snapshot', () => {
    const tree = renderer
      .create(<Display />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
