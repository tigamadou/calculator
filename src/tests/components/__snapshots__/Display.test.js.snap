// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display compoent tests It match with snapshot 1`] = `
<div
  className="display"
>
  0
</div>
`;
